app/src/main/java/org/greenstand/android/TreeTracker/managers/User.kt

```diff
@@ -0,0 +1,56 @@
+package org.greenstand.android.TreeTracker.managers
+
+import org.greenstand.android.TreeTracker.preferences.PrefKey
+import org.greenstand.android.TreeTracker.preferences.PrefKeys
+import org.greenstand.android.TreeTracker.preferences.Preferences
+import org.koin.core.context.GlobalContext
+
+class User(
+    private val preferences: Preferences
+) {
+
+    val isLoggedIn: Boolean
+        get() = planterCheckinId != -1L
+
+    var firstName: String?
+        get() = preferences.getString(FIRST_NAME_KEY)
+        set(value) = preferences.edit().putString(FIRST_NAME_KEY, value).apply()
+
+    var lastName: String?
+        get() = preferences.getString(LAST_NAME_KEY)
+        set(value) = preferences.edit().putString(LAST_NAME_KEY, value).apply()
+
+    var profilePhotoPath: String?
+        get() = preferences.getString(PROFILE_PHOTO_PATH_KEY)
+        set(value) = preferences.edit().putString(PROFILE_PHOTO_PATH_KEY, value).apply()
+
+    var organization: String?
+        get() = preferences.getString(ORG_NAME_KEY)
+        set(value) = preferences.edit().putString(ORG_NAME_KEY, value).apply()
+
+    var planterCheckinId: Long?
+        get() = preferences.getLong(PLANTER_CHECK_IN_ID_KEY)
+        set(value) = preferences.edit().putLong(PLANTER_CHECK_IN_ID_KEY, value ?: -1).apply()
+
+    var planterInfoId: Long?
+        get() = preferences.getLong(PLANTER_INFO_ID_KEY)
+        set(value) = preferences.edit().putLong(PLANTER_INFO_ID_KEY, value ?: -1).apply()
+            .also { preferences.setPlanterInfoId(value) }
+
+    var lastCheckInTimeInSeconds: Long?
+        get() = preferences.getLong(LAST_CHECK_IN_TIME_IN_KEY)
+        set(value) = preferences.edit().putLong(LAST_CHECK_IN_TIME_IN_KEY, value ?: -1).apply()
+
+    fun expireCheckInStatus() = preferences.clearPrefKeyUsage(BASE_KEY)
+
+    companion object {
+        private val BASE_KEY = PrefKeys.SESSION + PrefKey("info")
+        val PLANTER_CHECK_IN_ID_KEY = BASE_KEY + PrefKey("planter-check-in-id")
+        val PLANTER_INFO_ID_KEY = BASE_KEY + PrefKey("planter-info-id")
```

Comment:
It would be a bit more intuitive if we have `PLANTER_INFO_ID_KEY` derived from `UserPrefKey` and to omit line 38. This in combination with line 39 in `computePath` to use `UserPrefKey` will make it that much more cohesive.

Reply:
The User class holds the current user data **for the app**. So these preferences are not associated with an individual user.